src/components/Consents/Consents.test.js

```diff
@@ -0,0 +1,89 @@
+import React from 'react';
+import { render, screen } from '@testing-library/react';
+import '@testing-library/jest-dom';
+import { Provider } from 'react-redux';
+import configureStore from 'redux-mock-store';
+import thunk from 'redux-thunk';
+import Consent from 'components/Consents';
+import 'i18NextInit';
+
+const store = (
+  loading = false,
+  error = '',
+  definitions = [],
+  labels = [],
+  checked = []
+) => ({
+  consents: {
+    definitions,
+    labels,
+    checked,
+    loading,
+    error
+  },
+  publisherConfig: { publisherId: '' }
+});
+
+const middleware = [thunk];
+const mockStore = configureStore(middleware);
+
+describe('PaymentInfo component', () => {
+  test('should render GeneralErrorStyled (if loading = false)', async () => {
+    const { getByText } = render(
+      <Provider store={mockStore(store(false, 'noPublisherId'))}>
+        <Consent />
+      </Provider>
+    );
+
+    expect(
+      getByText(
+        /Unable to fetch terms & conditions. Publisher is not recognized/i
+      )
+    ).toBeInTheDocument();
+  });
+  test('should render GeneralErrorStyled (if loading = true)', async () => {
+    const { getByText } = render(
+      <Provider store={mockStore(store(true, 'noPublisherId'))}>
+        <Consent />
+      </Provider>
+    );
+
+    expect(
+      getByText(
+        /Unable to fetch terms & conditions. Publisher is not recognized/i
+      )
+    ).toBeInTheDocument();
+  });
+  test('should render Loader component', async () => {
+    const { getByTestId } = render(
+      <Provider store={mockStore(store(true))}>
+        <Consent />
+      </Provider>
+    );
+
+    expect(getByTestId('consents__loader')).toBeInTheDocument();
+  });
+  test('should render consents', async () => {
+    const definitions = [
```

Comment:
I'm not sure if we should stay with 'definitions' naming. This is a resource from our API that describes publisher consents, so I think naming it as 'publisherConsents' would be better (also with file names, slice,s etc)